routes/applicationRoutes.js

```diff
@@ -70,16 +73,30 @@ router.put('/:id?approve', async (req, res) => {
 			{ baseUser: id },
 			{ approved: true, rejected: false }
 		);
+
+		// Fetch application belonging to content creator
+		const application = await ApplicationModel.findOne({ baseUser: id });
+		if (!application) {
+			console.error('Application for content creator not found in database!');
+			throw new Error('E1005');
+		}
         
 		//send email to the user
 		await approveEmail(id);
 
+		// Save academic and work experience forms to database
+		await storeEducationAndExperienceFormsInDB(application);
+
 		//Return successful response
 		return res.status(200).json();
 
 	} catch(error) {
-		//If anything unexpected happens, throw error
-		return res.status(400).json({ 'error': errorCodes['E1003'] }); //Could not approve Content Creator
+		if (error.message === 'E1005') 
```

Comment:
error.code instead of error.message
Consider switch case as well with many checks on the same property

Reply:
> error.code instead of error.message

_No can do, Homer_  😟
The `Error` class has no `code` property, and I would have to add two more lines for each thrown error to assign said property to each object:
`const error = new Error(errorCodes['E1007']);`
`error.code = 'E1007';`
`throw error;`

I just don't think it's worth it here.

> Consider switch case as well with many checks on the same property

As you wish, milord: 
![image](https://github.com/user-attachments/assets/7ce1f534-4a0d-477f-a0f5-c5de557f9dc3)
Although, a silly ESLint indentation error forced me to align each `case` with the `switch` expression 😬